DDDEastAnglia.Tests/InMemoryRepository.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace DDDEastAnglia.Tests
+{
+    internal class InMemoryRepository<T>
+    {
+        private readonly Func<T, int> keyFunc;
+        private readonly ConcurrentDictionary<int, T> store = new ConcurrentDictionary<int, T>();
+
+        public InMemoryRepository(Func<T, int> keyFunc)
+        {
+            this.keyFunc = keyFunc;
```

Comment:
IMO, null checks do make sense because they assert the preconditions for using the class hold true. Yes, it's true to say that Ninject will b0rk before we ever get a chance to pass a null to most things, but we do also create a lot of stuff manually in our code base. 

More generally, I don't feel an application should be dependent on the behaviour of a third-party library in this way to be minimally functional. We should be able to remove Ninject and replace it with manual composition (or another IoC container) and still have everything work fine.